### DISCUSSION TITLE
Ideas for new features?

### DISCUSSION THREAD

**Discussion by evbacher** [evbacher/gd2md-html#181] [Ideas]
Hi all,
A couple of us are looking to enhance Docs to Markdown (in addition to addressing some outstanding bugs and feature requests).

Before we get started, wanted to ask you followers if you have any ideas that you haven't already noted in bug reports or feature requests. Can't guarantee that we'll implement all of them, but we'll certainly consider them.

A few ideas that we've been playing with and have implemented prototypes of:
- variable: single line or multi-line text blocks that can be replaced throughout the document by referencing the variable name (in conversion output, of course).
- definitions of important terms: Users can hover over marked terms and see the definition as a tool tip.
- acronym expansion: When acronym/abbreviation expansions are provided, we replace (in the conversion output) the first instance of the acronym with the expansion followed by the acronym. For example: Community Learning Center (CLC).
- comments: Allow comments in the Google Doc. These are not visible in the rendered output (either ignored or written as HTML commments.

Thanks,
evbacher

**Comment by rheaster** [comment]
I'm not sure if this falls under the 'variable' idea, but I often use gDocs to write for things to publish online. In this case some characters speak in different font types, flashbacks are in a different font, instant messaging is in another.
Would the code be able to recognize different fonts and append HTML of my choice (or even just tags with the font name in them for a fast find/replace)?

**Reply by evbacher** [reply]
It is possible to detect the font family of text. But this would add some complexity and overhead to the code. And it is a bit beyond the scope for the general use case of converting doc content to simple Markdown or HTML.

As you suggest, you can do this now with some markup of your own, then do some post processing to find/replace the variable value with the HTML that you want to insert. For example: startFlashback flashback text here endFlashback. etc.

Or, better yet, with vars, you could just set the value of the variable to the HTML you want:

.var flashback = &lt;span class="flashback"> 
.var endspan = &lt;/span>

and then in the doc: For example: .$flashback flashback text here .$endspan. etc.

**Reply by rheaster** [reply]
Thank you for your prompt reply! :D
I'm hoping to be able to just hit the Convert button and have it append the HTML, as I am hoping to keep the word count correct and the original gDoc text legible/streamlined. Is this something I could do if I made a copy of this extension? I am but a newbie when it comes to coding--most of my knowledge is from HTML in the 90s; making my own Neopets page look decent :P

I don't want to weigh down the main extension for others! I just prefer a 'do the hard work first (making the program) to avoid work later (individually inputting the code)'

Thanks again for your help, and if there's someone you know who could do this for me, for a fee or what have you, I'd love to get in contact with them.
